execution_engine_testing/tests/src/test/explorer/faucet.rs

```diff
@@ -1,77 +1,975 @@
+        installer_account,
+        FAUCET_INSTALLER_SESSION,
+        runtime_args! {ARG_ID => FAUCET_ID, ARG_AMOUNT => faucet_fund_amount },
     )
     .build();
 
-    let mut builder = InMemoryWasmTestBuilder::default();
+    builder
+        .exec(installer_session_request)
+        .expect_success()
+        .commit();
+
+    let faucet_install_cost = builder.last_exec_gas_cost();
+    assert_eq!(
+        faucet_install_cost.value().as_u64(),
+        EXPECTED_FAUCET_INSTALL_COST
+    );
+
+    let assigned_time_interval = 10_000u64;
+    let assigned_distributions_per_interval = 2u64;
+    let installer_set_variable_request = {
+        let deploy_item = DeployItemBuilder::new()
+            .with_address(installer_account)
+            .with_authorization_keys(&[installer_account])
+            .with_stored_session_named_key(
+                &format!("{}_{}", FAUCET_CONTRACT_NAMED_KEY, FAUCET_ID),
+                ENTRY_POINT_SET_VARIABLES,
+                runtime_args! {
+                    ARG_AVAILABLE_AMOUNT => Some(faucet_fund_amount),
+                    ARG_TIME_INTERVAL => Some(assigned_time_interval),
+                    ARG_DISTRIBUTIONS_PER_INTERVAL => Some(assigned_distributions_per_interval)
+                },
+            )
+            .with_empty_payment_bytes(runtime_args! {ARG_AMOUNT => *DEFAULT_PAYMENT})
+            .with_deploy_hash([3; 32])
+            .build();
+
+        ExecuteRequestBuilder::from_deploy_item(deploy_item).build()
+    };
+
+    builder
+        .exec(installer_set_variable_request)
+        .expect_success()
+        .commit();
+
+    let faucet_set_variables_cost = builder.last_exec_gas_cost();
+    assert_eq!(
+        faucet_set_variables_cost.value().as_u64(),
+        EXPECTED_FAUCET_SET_VARIABLES_COST
+    );
+
+    let user_fund_amount = U512::from(3_000_000_000u64);
+    let faucet_call_by_installer = {
+        let deploy_item = DeployItemBuilder::new()
+            .with_address(installer_account)
+            .with_authorization_keys(&[installer_account])
+            .with_stored_session_named_key(
+                &format!("{}_{}", FAUCET_CONTRACT_NAMED_KEY, FAUCET_ID),
+                ENTRY_POINT_FAUCET,
+                runtime_args! {ARG_TARGET => user_account, ARG_AMOUNT => user_fund_amount, ARG_ID => <Option<u64>>::None},
+            )
+            .with_empty_payment_bytes(runtime_args! {ARG_AMOUNT => user_fund_amount})
+            .with_deploy_hash([4; 32])
+            .build();
+
+        ExecuteRequestBuilder::from_deploy_item(deploy_item).build()
+    };
+
+    builder
+        .exec(faucet_call_by_installer)
+        .expect_success()
+        .commit();
+
+    let faucet_call_by_installer_cost = builder.last_exec_gas_cost();
+    assert_eq!(
+        faucet_call_by_installer_cost.value().as_u64(),
+        EXPECTED_FAUCET_CALL_BY_INSTALLER_COST
+    );
+
+    let faucet_contract_hash = builder
```

Comment:
```suggestion
    let faucet_contract_hash = get_faucet_contract_hash(&builder, installer_account);
```

Could we use the helper function here?

Reply:
Yes - good catch! :)

Done in [98e09fd](https://github.com/casper-network/casper-node/pull/2976/commits/98e09fd19990fba30486058426258f4fafe72349).